packages/react-atlas-core/src/avatar/Avatar.js

```diff
@@ -6,32 +6,64 @@ import themeable from 'react-themeable';
  *
  * **NOTE**: children will always take precedence over props passed into component.
  */
-const Avatar = ({children, icon, image, title, ...props}) => {
-  const theme = themeable(props.theme);
-  let kids = children;
-  if (React.Children.count(children) === 1 && typeof children === "string"){
-    kids = <span {...theme(4, 'letter')}>{children[0]}</span>;
-  }
+ class Avatar extends React.Component {
+   constructor(props) {
+     super(props);
 
-  let avatar = null;
+     let image;
 
-  if (typeof image === "string") {
-    avatar = <img src={image} title={title} {...theme(3, 'image')} />;
-  } else if (image){
-    avatar = image;
-  } else if (icon){
-    avatar = icon;
-  } else if (title) {
-    avatar = <span {...theme(2, 'letter')}>{title[0]}</span>;
-  }
+     if(this.props.image === undefined) {
+       if(this.props.defaultImage === undefined) {
+         image = null;
+       } else {
+         image = this.props.defaultImage;
+       }
+     } else {
+       image = this.props.image;
+     }
 
-  return (
-    <div {...props} {...theme(1, 'avatar')}>
-      {kids}
-      {avatar}
-    </div>
-  );
-};
+     this.state = {image: image, title: this.props.title, icon: this.props.icon};
+   }
+
+   handleBadImage() {
+     if(this.props.defaultImage === this.state.image) {
+        this.setState({image: null});
+        return;
+     }
+
+     this.setState({image: this.props.defaultImage});
+   }
+
+   render() {
+     const theme = themeable(this.props.theme);
+     var children = this.props.children;
+     let kids = children;
+     if (React.Children.count(children) === 1 && typeof children === "string"){
+       kids = <span {...theme(4, 'letter')}>{children[0]}</span>;
+     }
+
+     let avatar = null;
+     var icon = this.state.icon;
+     var image = this.state.image;
+     var title = this.state.title;
```

Comment:
if we take icon and title out of props, we can use es6 destructring above where we defined `children` and so something like:
`var {children, icon, title} = this.props;`